src/sandbox/components/Worksheet.svelte

```diff
@@ -97,6 +99,107 @@
       return true;
     }
 
+    if (modifiers === 4 && (e.code === "BracketLeft" || e.code === "BracketRight" || e.code === "Backslash")) {
+
+      const normalizedSelection = normalize(selection);
+      [rowStart, colStart, rowEnd, colEnd] = normalizedSelection;
+
+      const [rows, cols] = [rowEnd - rowStart, colEnd - colStart];
+      const [isColumn, isRow] = [!cols, !rows];
+      let isVector = isRow || isColumn; // Check array dimension
+
+      if (isRow && isColumn) return true; // No action on single cell
+
+      const selectionSet = [...query(blocks).getInRange(normalizedSelection)];
+      if (!selectionSet.length) return true; // Return if selection is empty
+
+      for (let i = 0; i < selectionSet.length; i++) {
+        // Return if selection contains formula
+        if (blocks.get(selectionSet[i]).node) return true;
+      }
+
+      let result = [];
+      let dataType;
+
+      if (e.code === "BracketLeft") {
+        if (isVector) {
+          result = makeVector(rowStart, colStart, isColumn ? rows : cols, isColumn);
+        }
+        else {
+          for (let i = 0; i <= rows; i++) {
+            result[i] = makeVector(rowStart + i, colStart, cols, isColumn);
```

Comment:
better pass false instead of isColumn explicitly here

Reply:
Fixed.